lib/features/thread/data/network/thread_api.dart

```diff
@@ -71,16 +75,32 @@ class ThreadAPI {
       .build()
       .execute();
 
-    final resultList = result.parse<GetEmailResponse>(
-        getEmailInvocation.methodCallId, GetEmailResponse.deserialize);
+    final responseOfGetEmailMethod = result.parse<GetEmailResponse>(
+      getEmailInvocation.methodCallId,
+      GetEmailResponse.deserialize,
+    );
 
-    if (sort != null && resultList != null) {
-      for (var comparator in sort) {
-        resultList.sortEmails(comparator);
-      }
+    final responseOfQueryEmailMethod = result.parse<QueryEmailResponse>(
+      queryEmailInvocation.methodCallId,
+      QueryEmailResponse.deserialize,
+    );
+
+    List<Email>? emailList;
+
+    if (responseOfGetEmailMethod?.list.isNotEmpty == true &&
+        responseOfQueryEmailMethod?.ids.isNotEmpty == true) {
+      log('ThreadAPI::getAllEmail: QUERY_EMAIL_IDS = ${responseOfQueryEmailMethod?.ids}');
+      final listSortedEmail = responseOfGetEmailMethod!.list
+        .sortingByOrderOfIdList(responseOfQueryEmailMethod!.ids.toList());
```

Comment:
for the case of `loadMore` we don't need to change anything?. still base on `recievedAt` no need to base on position?

Reply:
Yes, We've already handled that. Here we just change the `sort` function to use it via `Email/query`